bson/timestamp.py

```diff
@@ -54,7 +55,9 @@ def __init__(self, time, inc):
         """
         if isinstance(time, datetime.datetime):
             if time.utcoffset() is not None:
```

Comment:
Suggest rewriting to remove the duplicate utcoffset() call and redudant assert:
```python
offset = time.utcoffset()
if offset is not None:...
```

Reply:
Done